src/main/java/com/endava/client/Client.java

```diff
@@ -0,0 +1,44 @@
+package com.endava.client;
+
+import com.endava.repository.Student;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Arrays;
+
+public class Client {
+
+    public static void main(String[] args) {
+
+        RestTemplate restTemplate = new RestTemplate();
+
+        // find student with id
+        Student student = restTemplate.getForObject("http://localhost:8080/student/id?id=123200007", Student.class);
```

Comment:
This URL scheme /id?id=... doesnt make sense, why do you need to have "id" as both path param and query param

Reply:
It is reversed.